setup.cfg

```diff
@@ -1,7 +1,7 @@
 [flake8]
 ignore = E402
 exclude = test/*
-max-complexity = 15
+max-complexity = 17
```

Comment:
Yeah, `read_config_file()` is pushing the complexity limit.

I split off the helper function `_read_etcd_config_key()` to try and keep the complexity under control, but even then the small
```
if 'ETCD_MACHINES' in os.environ`:
    ...
```
block I added bumped the complexity a couple points.

Was gonna circle back later and try to break it up further in a separate cleanup PR.